native/cocos/bindings/manual/Worker.cpp

```diff
@@ -0,0 +1,619 @@
+/**
+ * Created by ihowe@outlook.com on 2023/5/31.
+ * Worker on Native
+ */
+
+#include <chrono>
+#include <thread>
+#include <sstream>
+#include <atomic>
+
+#include "Worker.h"
+// include cocos
+#include <cocos/base/Macros.h>
+#include <cocos/base/Log.h>
+#include <cocos/platform/FileUtils.h>
+#include <cocos/bindings/jswrapper/config.h>
+
+#define __NANOSECONDS_PER_SECOND 1000000000
+#define __NANOSECONDS_60FPS      16666667L
+#define __prefererredNanosecondsPerFrame    static_cast<long>(1.0 / 60 * __NANOSECONDS_PER_SECOND)
+
+#define USE_HANDLE_SCOPE v8::HandleScope handle_scope(isolate)
+#define FILE_UTILS cc::FileUtils::getInstance()
+
+#if SCRIPT_ENGINE_TYPE == SCRIPT_ENGINE_V8
+
+// include v8
+#include "libplatform/libplatform.h"
+#include "v8.h"
+
+/**
+ * setTimeout / clearTimeout /setInterval /clearInterval / self / window
+ */
+const std::string runtimejs = R"(
+(function(root){root.window=root;root.self=root;let scheduleDataModel={ID:1,timeoutInfos:null}
+scheduleDataModel.timeoutInfos=new Map();let TimeInfo=function(cb,delay,isRepeat,target,args){this.cb=cb;this.id=++scheduleDataModel.ID;this.start=Date.now();this.delay=delay;this.isRepeat=isRepeat;this.target=target;this.args=args;};function handlerTimerInfos(nowMilliSeconds){if(scheduleDataModel.timeoutInfos.size>0){let infos=Array.from(scheduleDataModel.timeoutInfos.values());for(let info of infos){if(nowMilliSeconds-info.start>=info.delay){if(typeof info.cb==='string'){Function(info.cb)();}else if(typeof info.cb==='function'){info.cb.apply(info.target,info.args);}
+if(info.isRepeat){info.start=nowMilliSeconds;}else{info.cb=null;info.target=null;info.args=null;scheduleDataModel.timeoutInfos.delete(info.id);}}}}}
+function createTimeoutInfo(prevFuncArgs,isRepeat){let cb=prevFuncArgs[0];if(!cb){console.error("createTimeoutInfo doesn't pass a callback ...");return 0;}
+let delay=prevFuncArgs.length>1?prevFuncArgs[1]:0;let args;if(prevFuncArgs.length>2){args=Array.prototype.slice.call(prevFuncArgs,2);}
+let info=new TimeInfo(cb,delay,isRepeat,root,args);scheduleDataModel.timeoutInfos.set(info.id,info);return info.id;}
+root.setTimeout=function(cb){return createTimeoutInfo(arguments,false);};root.clearTimeout=function(id){scheduleDataModel.timeoutInfos.delete(id);};root.setInterval=function(cb){return createTimeoutInfo(arguments,true);};root.clearInterval=root.clearTimeout;root["$globalTick"]=function(nowMilliSeconds){handlerTimerInfos(nowMilliSeconds);}})(globalThis)
```

Comment:
Where does this code come from?
If there is a bug in this code, how could we fix it since it was minified.